### ISSUE TITLE
Doesn't work with rollup >= 1.27.0

### ISSUE THREAD

**Issue by targumon** [rollup/plugins#79]
- Rollup Plugin Name: replace
- Rollup Plugin Version: 2.2.1
- Rollup Version: 1.27.8
- Operating System (or Browser): MacOS
- Node Version: 10.15.3

### How Do We Reproduce?

Created a clean project: `npm i -y`, followed by `npm i -d rollup @rollup/plugin-replace`, created a small js file with just console.log in it, using the exact rollup.config.js file described under 'Usage' in https://github.com/rollup/plugins/tree/master/packages/replace

### Expected Behavior
string replaced in the output

### Actual Behavior
string not replaced in the output
warning in terminal: _(!) The "transform" hook used by the output plugin replace is a build time hook and will not be run for that plugin. Either this plugin cannot be used as an output plugin, or it should have an option to configure it as an output plugin._

I suspect the problem is related to https://github.com/rollup/rollup/pull/3218
and so I'm guessing it's the documentation of the plugin that needs to be fixed, not the plugin itself.

**Comment by lukastaegert** [comment]
Are you sure you placed the `plugins` option in the correct place? This error can only occur if it is placed inside the `output` object. In the example you linked, it is placed outside the `output` object.

**Comment by lukastaegert** [comment]
Though it might make sense to be able to use this plugin as an output plugin, it is not supported and also would not really make sense as then it would not benefit from tree-shaking.

**Comment by shellscape** [comment]
@targumon as the issue template mentions, we need a minimal reproduction to triage this. Without that, there's not much we can do. 

>   Issues without minimal reproductions will be closed! Please provide one by:
>   1. Using the REPL at https://rollupjs.org/repl/, or
>   2. Work to isolate the problem and provide the exact steps in this issue, or
>   3. Provide a minimal repository link (Read https://git.io/fNzHA for instructions).
>      These may take more time to triage than the other options.

**Comment by targumon** [comment]
Oh, I feel so stupid. I'm not even sure now where did I copy my bad `rollup.config.js` from (with `plugins` inside `output` instead of adjacent to it).

Thanks and my apologies for the trouble!

**Comment by targumon** [comment]
Found the source of my mistake: before realizing I also need the _replace_ plugin, the first one I used was _terser_, which is an output plugin... So my initial config had `plugins` inside `output` and I naturally thought that that's the place for **all** plugins.

TL;DR (for whoever gets here from search engines):
There are "regular" plugins and there are output plugins, so your _rollup.config.js_ may contain the `plugins` option more than once:
```
export default {
  input: 'index.js',
  output: {
    dir: 'dist',
    format: 'iife', // amd cjs system esm iife umd
    plugins: [
      terser()
    ]
  },
  plugins: [
    replace({'process.env.NODE_ENV': JSON.stringify('production')})
  ]
}
```

More details here: https://rollupjs.org/guide/en/#using-output-plugins

**Comment by marcofugaro** [comment]
I also have this warning when using the replace plugin as an output plugin like this

```js
export default {
  input: 'index.js',
  output: {
    dir: 'dist',
    format: 'iife', // amd cjs system esm iife umd
    plugins: [
      replace({'process.env.NODE_ENV': JSON.stringify('production')}),
      terser(),
    ]
  }
}
```

In my case, I actually need it on the output bundle and it is allowed since #55. So why there is that warning?
